### PR TITLE
adding support for module manifest schema

### PR DESCRIPTION
**Pull request by dgraeber**
*Issue #, if available:*
#612 
*Description of changes:*
Adding support to fetch only the module manifest schema

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: seedfarmer/mgmt/build_info.py [hunk 1]
```diff
@@ -40,5 +41,10 @@ def get_build_env_params(build_ids: List[str], session: Optional[Session] = None
...
 
-def get_manifest_schema() -> Dict[str, Any]:
-    return DeploymentManifest.model_json_schema()
+def get_manifest_schema(type: str) -> Optional[Dict[str, Any]]:
+    if "deployment" == type.lower():
+        return DeploymentManifest.model_json_schema()
+    elif "module" == type.lower():
+        return ModuleManifest.model_json_schema()
+    else:
+        return None
```
Comment (LeonLuttenberger) [inline seedfarmer/mgmt/build_info.py:50]: Minor nitpick: We could just raise an error here, and that way wouldn't need to change the return type of the function.

File: seedfarmer/cli_groups/_list_group.py [hunk 2]
```diff
@@ -779,8 +779,20 @@ def list_build_env_params(
...
+@click.option(
+    "--type",
+    "-t",
+    help="Either 'deployment' or 'module' can be used, default is `deployment`",
+    required=False,
+    default="deployment",
+)
+def list_manifest_schema(type: str) -> None:
+    if type not in ["deployment", "module"]:
+        print_bolded(message="Only 'deployment' or 'module' can be passed in", color="yellow")
```
Comment (LeonLuttenberger) [inline seedfarmer/cli_groups/_list_group.py:796]: We can use [click.Choice](https://click.palletsprojects.com/en/8.1.x/options/#choice-options) to perform this validation for us.

File: test/unit-test/test_mgmt_build_info.py [hunk 3]
```diff
@@ -67,4 +67,4 @@ def test_get_build_params(mocker):
 @pytest.mark.mgmt
 @pytest.mark.mgmt_build_info
 def test_validate_group_parameters():
-    bi.get_manifest_schema()
```
Comment (LeonLuttenberger) [inline test/unit-test/test_mgmt_build_info.py:70]: Do we maybe want another unit test here for `type="module"`?

Comment (dgraeber) [reply test/unit-test/test_mgmt_build_info.py:70]: yep....can do

File: seedfarmer/cli_groups/_list_group.py (hunk 2 above)
Comment (dgraeber) [reply seedfarmer/cli_groups/_list_group.py:796]: Will research

File: seedfarmer/mgmt/build_info.py (hunk 1 above)
Comment (dgraeber) [reply seedfarmer/mgmt/build_info.py:50]: hmmm...will check

**Review by LeonLuttenberger — APPROVED** [review]

**Pull request merged**